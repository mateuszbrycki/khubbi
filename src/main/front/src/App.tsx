import React from 'react';
import AuthorizationContainer from './authorization/containers/AuthorizationContainer'
import './App.css';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import {Col, Row} from "react-bootstrap";

export interface AppActionsProps {
}

export interface AppProps {
}


function App() {
  return (
      <>
      <Navbar expand="lg" className="bg-body-tertiary">
        <Container>
          <Navbar.Brand href="#home">bookkeeper</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="me-auto">
              <Nav.Link href="#home">Home</Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
      <AuthorizationContainer/>
      </>
  );
}

export default App;
