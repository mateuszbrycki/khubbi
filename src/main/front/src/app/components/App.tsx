import React from "react";
import {history} from "../../store/store";
import {Link, Route, Routes} from "react-router-dom";
import {HistoryRouter as Router} from "redux-first-history/rr6";
import AuthorizationContainer from "../../authorization/containers/AuthorizationContainer";
import DashboardContainer from "../../dashboard/containers/DashboardContainer";
import AuthenticationAwareGate from "../../common/AuthenticationAwareGate";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";

export interface AppProps {
    readonly isAuthenticated: boolean
    readonly authenticationCheckInterval: number
}

export interface AppActionProps {
    readonly logoutUser: () => void
    readonly checkIfUserIsAuthenticated: () => void
}

const App: React.FC<AppProps & AppActionProps> = (props) => {

    const {isAuthenticated, authenticationCheckInterval, logoutUser, checkIfUserIsAuthenticated} = props

    React.useEffect(() => {

        if (isAuthenticated) {
            const interval = setInterval(() => {
                checkIfUserIsAuthenticated()
            }, authenticationCheckInterval);

            return () => clearInterval(interval);
        }

    }, [isAuthenticated])

    return <> <Router history={history}>
        <Navbar expand="lg" className="bg-body-tertiary">
            <Container>
                <Navbar.Brand href="#home">bookkeeper</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav"/>
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="me-auto">
                        <Nav.Link> <Link to={`/`}>Home</Link></Nav.Link>
                        {isAuthenticated ? <></> : <Nav.Link> <Link to={`/login`}>Login</Link></Nav.Link>}
                    </Nav>
                    <Nav className="ml-auto">
                        {isAuthenticated ? <Nav.Link> <Link to="/" onClick={event => {
                            event.preventDefault();
                            logoutUser();
                        }}>Logout</Link></Nav.Link> : <></>}
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
        <Container>
            <Routes>
                <Route path="/" element={
                    <AuthenticationAwareGate isAuthenticated={isAuthenticated}
                                             redirectToIfLogged={"/dashboard"}>
                        <h1>Welcome in bookkeeper</h1>
                    </AuthenticationAwareGate>
                }/>
                <Route path="/login"
                       element={
                           <AuthenticationAwareGate isAuthenticated={isAuthenticated}
                                                    redirectToIfLogged={"/dashboard"}>
                               <AuthorizationContainer/>
                           </AuthenticationAwareGate>
                       }/>
                <Route path="/dashboard"
                       element={
                           <AuthenticationAwareGate isAuthenticated={isAuthenticated}
                                                    redirectToIfNotLogged={"/login"}>
                               <DashboardContainer/>
                           </AuthenticationAwareGate>}/>
            </Routes>
        </Container>
    </Router>
    </>

}

export default App