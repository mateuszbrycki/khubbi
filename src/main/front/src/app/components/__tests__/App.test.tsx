import ShallowRenderer from "react-test-renderer/shallow";
import App from "../App";

describe('application component', () => {
    describe('user is authenticated', () => {
        const isAuthenticated = true

        it('renders correctly', () => {
            const renderer = ShallowRenderer.createRenderer()
            renderer.render(<App logoutUser={() => null} isAuthenticated={isAuthenticated}/>)
            expect(renderer.getRenderOutput()).toMatchSnapshot()
        });
    })
    describe('user is not authenticated', () => {
        const isAuthenticated = false

        it('renders correctly', () => {
            const renderer = ShallowRenderer.createRenderer()
            renderer.render(<App logoutUser={() => null} isAuthenticated={isAuthenticated}/>)
            expect(renderer.getRenderOutput()).toMatchSnapshot()
        });
    })
})
