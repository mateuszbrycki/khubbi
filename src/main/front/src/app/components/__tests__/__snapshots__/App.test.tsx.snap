// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application component user is authenticated renders correctly 1`] = `
<React.Fragment>
   
  <HistoryRouter
    history={
      {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": undefined,
        "listen": [Function],
        "listenObject": false,
        "location": {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar
      className="bg-body-tertiary"
      expand="lg"
    >
      <Container>
        <NavbarBrand
          href="#home"
        >
          bookkeeper
        </NavbarBrand>
        <NavbarToggle
          aria-controls="basic-navbar-nav"
        />
        <NavbarCollapse
          id="basic-navbar-nav"
        >
          <Nav
            className="me-auto"
          >
            <NavLink>
               
              <Link
                to="/"
              >
                Home
              </Link>
            </NavLink>
            <React.Fragment />
          </Nav>
          <Nav
            className="ml-auto"
          >
            <NavLink>
               
              <Link
                onClick={[Function]}
                to="/"
              >
                Logout
              </Link>
            </NavLink>
          </Nav>
        </NavbarCollapse>
      </Container>
    </Navbar>
    <Container>
      <Routes>
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={true}
              redirectToIfLogged="/dashboard"
            >
              <h1>
                Welcome in bookkeeper
              </h1>
            </AuthenticationAwareGate>
          }
          path="/"
        />
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={true}
              redirectToIfLogged="/dashboard"
            >
              <Memo(Connect(EntryForms)) />
            </AuthenticationAwareGate>
          }
          path="/login"
        />
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={true}
              redirectToIfNotLogged="/login"
            >
              <Memo(Connect(Dashboard)) />
            </AuthenticationAwareGate>
          }
          path="/dashboard"
        />
      </Routes>
    </Container>
  </HistoryRouter>
</React.Fragment>
`;

exports[`application component user is not authenticated renders correctly 1`] = `
<React.Fragment>
   
  <HistoryRouter
    history={
      {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": undefined,
        "listen": [Function],
        "listenObject": false,
        "location": {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar
      className="bg-body-tertiary"
      expand="lg"
    >
      <Container>
        <NavbarBrand
          href="#home"
        >
          bookkeeper
        </NavbarBrand>
        <NavbarToggle
          aria-controls="basic-navbar-nav"
        />
        <NavbarCollapse
          id="basic-navbar-nav"
        >
          <Nav
            className="me-auto"
          >
            <NavLink>
               
              <Link
                to="/"
              >
                Home
              </Link>
            </NavLink>
            <NavLink>
               
              <Link
                to="/login"
              >
                Login
              </Link>
            </NavLink>
          </Nav>
          <Nav
            className="ml-auto"
          >
            <React.Fragment />
          </Nav>
        </NavbarCollapse>
      </Container>
    </Navbar>
    <Container>
      <Routes>
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={false}
              redirectToIfLogged="/dashboard"
            >
              <h1>
                Welcome in bookkeeper
              </h1>
            </AuthenticationAwareGate>
          }
          path="/"
        />
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={false}
              redirectToIfLogged="/dashboard"
            >
              <Memo(Connect(EntryForms)) />
            </AuthenticationAwareGate>
          }
          path="/login"
        />
        <Route
          element={
            <AuthenticationAwareGate
              isAuthenticated={false}
              redirectToIfNotLogged="/login"
            >
              <Memo(Connect(Dashboard)) />
            </AuthenticationAwareGate>
          }
          path="/dashboard"
        />
      </Routes>
    </Container>
  </HistoryRouter>
</React.Fragment>
`;
